src/client/pfs/pfs.proto

```diff
@@ -495,6 +500,17 @@ message GetTarRequest {
   File file = 1;
 }
 
+message GetTarConditionalRequest {
+  File file = 1;
+  bool skip = 2;
+}
+
+message GetTarConditionalResponse {
+  FileInfoNewStorage file_info = 1;
+  bytes data = 2;
+  bool done = 3;
```

Comment:
What's the need for the `done` field? Can't doneness just be indicated by the stream ending?

Reply:
Done indicates the end of a file's content, not the end of the stream (refer to the protocol in the PR comment). `end_of_file` or something similar might be a better name for this.

Reply:
I see, I think `end_of_file` would be a better name. It's still technically redundant right, because when `done` is true it means either you're at the end of the stream, or the next message is going to have a different `file_info` than the one you just got? I'm cool with leaving the field in to be more explicit, we take the opposite approach in `PutFile` as it exists right now and it's worked fine. It's a little less explicit, but it also avoids some error cases, such as if the code switches to a different `file_info` and doesn't set `done` to true, or vice versa.

Reply:
I think being explicit will ultimately result with cleaner code. Right now, I am able to have an abstraction around reading a particular file that does not require me to keep track of the last message encountered (the next file info that will be needed for setting up the next callback). I could make the eof be indicated by an empty message because we will never have a zero size read due to the tar headers?

Reply:
I like the idea of making it explicit with an eof. I'm cool with having an explicit parameter called `eof` or `end_of_file` here, or having it be implicit, whichever seems better to you. The only thing I'm against is using `done` to indicate the end of a file's content since I think it's ambiguous between the end of a file and the end the stream as a whole.